### PR TITLE
Support new file lock configuration for buckets

### PR DESCRIPTION
**Pull request by mszumocki-reef**
* support isFileLockEnabled for buckets
* support defaultRetention for buckets
* fix acceptance tests breaking when new response fields are added to the API

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-16.04, macos-10.15]
```
Comment (mlech-reef) [inline .github/workflows/cd.yml:20]: It will cause that the provider won't work on older systems. Please revert

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ env:
 
 jobs:
   lint:
-    runs-on: ubuntu-16.04
```
Comment (mlech-reef) [inline .github/workflows/ci.yml:17]: Use reverted version

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+* support isFileLockEnabled for buckets
```
Comment (mlech-reef) [inline CHANGELOG.md:10]: We used to start an item with a capital letter here, in SDK, and in CLI. Please follow that convention

File: GNUmakefile [hunk 4]
```diff
@@ -42,7 +42,7 @@ lint: _pybindings
 
 testacc: _pybindings
 	@chmod +rx python-bindings/dist/py-terraform-provider-b2
-	TF_ACC=1 go test ./${NAME} -v -count 1 -parallel 4 -timeout 120m $(TESTARGS)
+	TF_ACC=1 go test ./${NAME} -v -count 1 -parallel 4 -v -timeout 120m $(TESTARGS)
```
Comment (mlech-reef) [inline GNUmakefile:45]: `-v` is already passed

File: b2/client.go [hunk 5]
```diff
@@ -43,8 +43,8 @@ type Client struct {
 }
 
 func (c Client) apply(name string, op string, input map[string]interface{}) (map[string]interface{}, error) {
-	log.Printf("[TRACE] Executing pybindings for '%s' and '%s' operation\n", name, op)
-	log.Printf("[TRACE] Input for pybindings: %+v\n", input)
+	log.Printf("[DEBUG] Executing pybindings for '%s' and '%s' operation\n", name, op)
```
Comment (mlech-reef) [inline b2/client.go:46]: Why?

File: python-bindings/b2_terraform/provider_tool.py [hunk 6]
```diff
@@ -12,10 +12,11 @@
 import hashlib
 import os
 import sys
+import traceback
 
 from class_registry import ClassRegistry
 from humps import camelize, decamelize
-from b2sdk.v1 import EncryptionAlgorithm, EncryptionMode, EncryptionSetting
+from b2sdk.v1 import EncryptionAlgorithm, EncryptionMode, EncryptionSetting, BucketRetentionSetting
```
Comment (mlech-reef) [inline python-bindings/b2_terraform/provider_tool.py:19]: Order alphabetically

File: python-bindings/b2_terraform/provider_tool.py [hunk 7]
```diff
@@ -26,6 +27,38 @@ def change_keys(obj, converter):
     return {converter(k).replace('__', '_'): v for k, v in obj.items()}
 
 
+def convert_json_to_go(obj: dict, keys_to_keep: dict):
+    """
```
Comment (mlech-reef) [inline python-bindings/b2_terraform/provider_tool.py:31]: Add a header line, e.g.: `Convert dict representation to follow Terraform Go SDK rules.`

File: python-bindings/b2_terraform/provider_tool.py [hunk 8]
```diff
@@ -165,21 +198,36 @@ def resource_create(
...
+            name=bucket_name,
+            bucket_type=bucket_type,
+            bucket_info=bucket_info,
+            cors_rules=cors_rules,
+            file_lock_configuration=file_lock_configuration,
+            default_server_side_encryption=default_server_side_encryption,
+            lifecycle_rules=lifecycle_rules,
         )
-        return self._postprocess(**bucket.as_dict())
+        # default retention can only be set with update_bucket, not create_bucket :(
```
Comment (mlech-reef) [inline python-bindings/b2_terraform/provider_tool.py:215]: WTF

File: python-bindings/b2_terraform/provider_tool.py [hunk 9]
```diff
@@ -340,16 +383,77 @@ def _preprocess(self, **kwargs):
...
-            server_side_encryption = []
-
-        return {
-            'serverSideEncryption': server_side_encryption,
-            **kwargs,
-        }
+        return convert_json_to_go(kwargs, FILE_VERSION_KEYS)
+
+
+SERVER_SIDE_ENCRYPTION = {
```
Comment (mlech-reef) [inline python-bindings/b2_terraform/provider_tool.py:389]: Can we move it to a separate file?

File: python-bindings/b2_terraform/provider_tool.py [hunk 10]
```diff
@@ -26,6 +27,38 @@ def change_keys(obj, converter):
     return {converter(k).replace('__', '_'): v for k, v in obj.items()}
 
 
+def convert_json_to_go(obj: dict, keys_to_keep: dict):
```
Comment (mlech-reef) [inline python-bindings/b2_terraform/provider_tool.py:30]: Maybe we can completely replace `change_keys` function with this one?

**Review by mlech-reef — CHANGES_REQUESTED** [review]

File: python-bindings/b2_terraform/provider_tool.py (hunk 10 above)
Comment (mszumocki-reef) [reply python-bindings/b2_terraform/provider_tool.py:30]: Not yet, as change_keys is also used in preprocess to convert to camel case.

File: python-bindings/b2_terraform/provider_tool.py (hunk 8 above)
Comment (mszumocki-reef) [reply python-bindings/b2_terraform/provider_tool.py:215]: Yes :(

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mszumocki-reef) [reply .github/workflows/ci.yml:17]: This ubuntu version is deprecated in GitHub actions...

File: b2/client.go (hunk 5 above)
Comment (mszumocki-reef) [reply b2/client.go]: Reverted

File: python-bindings/b2_terraform/provider_tool.py (hunk 6 above)
Comment (mszumocki-reef) [reply python-bindings/b2_terraform/provider_tool.py]: Fixed

File: python-bindings/b2_terraform/provider_tool.py (hunk 7 above)
Comment (mszumocki-reef) [reply python-bindings/b2_terraform/provider_tool.py:32]: Added

File: python-bindings/b2_terraform/provider_tool.py (hunk 9 above)
Comment (mszumocki-reef) [reply python-bindings/b2_terraform/provider_tool.py]: Moved

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mlech-reef) [reply .github/workflows/ci.yml]: So we should have a ticket to use docker runner. I will create one

Comment (mlech-reef) [reply .github/workflows/ci.yml]: #16 16

**Review by mlech-reef — APPROVED** [review]

**Pull request merged**